iceoryx_posh/source/popo/ports/server_port_user.cpp

```diff
@@ -39,15 +38,26 @@ ServerPortUser::MemberType_t* ServerPortUser::getMembers() noexcept
     return reinterpret_cast<MemberType_t*>(BasePort::getMembers());
 }
 
-
-cxx::expected<cxx::optional<const RequestHeader*>, ChunkReceiveResult> ServerPortUser::getRequest() noexcept
+cxx::expected<const RequestHeader*, ChunkReceiveResult> ServerPortUser::getRequest() noexcept
 {
-    return cxx::success<cxx::optional<const RequestHeader*>>(cxx::nullopt_t());
+    auto getChunkResult = m_chunkReceiver.tryGet();
+
+    if (getChunkResult.has_error())
+    {
+        return cxx::error<ChunkReceiveResult>(getChunkResult.get_error());
+    }
+
+    return cxx::success<const RequestHeader*>(static_cast<const RequestHeader*>(getChunkResult.value()->userHeader()));
 }
 
-void ServerPortUser::releaseRequest(const RequestHeader* const /*requestHeader*/) noexcept
+void ServerPortUser::releaseRequest(const RequestHeader* const requestHeader) noexcept
 {
-    /// @todo
+    cxx::Ensures(requestHeader != nullptr && "requestHeader must not be a nullptr");
```

Comment:
Could you please remove this. I would only print out an error log message since we can ensure here that we never reach an inconsistent state. Or maybe a call to the error handler with severity moderate?

Reply:
I will remove it from here, but the `ChunkReceiver` will call the error handler with error level `SEVERE` anyway.

Reply:
I just remembered why I did the check here instead of letting the next layer call the error handler. The problem is that the pointer will be de-referenced before `release` is called and therefore the check must remain here. I can create a `cxx::error` here but still with error level `SEVERE`. What do you think?

Reply:
@elfenpiff ping :)